src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -274,17 +290,13 @@ void EnzymesSelectorWidget::sl_filterTextChanged(const QString& filterText) {
 }
 
 void EnzymesSelectorWidget::updateStatus() {
-    int nChecked = 0;
-    QStringList checkedNamesList;
-    for (int i = 0, n = tree->topLevelItemCount(); i < n; i++) {
-        auto gi = static_cast<EnzymeGroupTreeItem*>(tree->topLevelItem(i));
-        nChecked += gi->checkedEnzymes.size();
-        foreach (const EnzymeTreeItem* ci, gi->checkedEnzymes) {
-            checkedNamesList.append(ci->enzyme->id);
-        }
+    QString checkedNamesListString;
+    int nChecked = gatherCheckedNamesListString(checkedNamesListString);
+    if (checkedNamesListString.size() > 50'000) {
+        checkedEnzymesEdit->setPlainText(tr("Too many sites selected to show, click \"Save selection\" to export them to the separate file"));
```

Comment:
"Too many" sounds like an error. Show the count instead.

Reply:
OK